cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -347,8 +363,9 @@ protected void setNetworkConfigs(Compute compute, GceClusterConfig.Builder clust
     try {
       systemNetwork = DataprocUtils.getSystemNetwork(
           (url) -> (HttpURLConnection) url.openConnection());
-    } catch (IllegalArgumentException e) {
+    } catch (DataprocRuntimeException e) {
       // expected when not running on GCP, ignore
+      LOG.error("Unable to get the system network.", e);
```

Comment:
I don't think you would need to do this. It would be just noise not on GCP. At least not as error with the trace. Anyway you would have "Unable to automatically detect a network" below. It would also be incorrect to report it as pipeline failure as it's handled and does not nesessarily leads to a failure (e.g. when user specified network). Same considerations below

Reply:
done, changed to trace.